### PR TITLE
Fix invalid response id after closing file in blocking context

### LINKED ISSUE #100: Invalid response id after closing file in blocking context
With https://github.com/openssh-rust/openssh-sftp-client/pull/98, the problem fixed by https://github.com/openssh-rust/openssh-sftp-client/pull/77 will exist. It occurs in action https://github.com/apache/incubator-opendal/actions/runs/5798363627/job/15715936917?pr=2815.

Maybe we need to find a better solution.

### PR DESCRIPTION
**Pull request by silver-ymz**
Close #100

### CODE REVIEW

File: src/sftp.rs [hunk 1]
```diff
@@ -146,6 +147,7 @@ impl Sftp {
                 options.get_max_pending_requests(),
                 auxiliary,
                 options.get_tokio_compat_file_write_limit(),
+                Handle::current(),
```
Comment (NobodyXu) [inline src/sftp.rs:150]: I would like this to be put into `Sftp::connect` to avoid monomorphization generating duplicate code.

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: src/sftp.rs (hunk 1 above)
Comment (silver-ymz) [reply src/sftp.rs]: changed in new commit

**Review by NobodyXu — APPROVED** [review]

**Comment by NobodyXu** [comment]
Thank you @silver-ymz !

**Comment by silver-ymz** [comment]
I also update CI to avoid it running twice in PR.

**Comment by silver-ymz** [comment]
Should we release it right now?

**Comment by NobodyXu** [comment]
> Should we release it right now?

Opened #102 , waiting for review

**Pull request merged**